### DISCUSSION TITLE
Can't change starred album cover photo: "No query results for model [App\Models\Album] starred"

### DISCUSSION THREAD

**Discussion by hess8** [LycheeOrg/Lychee#2428] [Bugs]
### Lychee version

v5.3.0

### Which PHP version are you using?

PHP 8.3

### Detailed description of the problem

Trying to set any photo as album cover photo in starred folder gives the error: "No query results for model [App\Models\Album] starred"

In other albums it works

Here is the full output: [Photos.pdf](https://github.com/LycheeOrg/Lychee/files/15241151/Photos.pdf)

### Steps to reproduce the issue

Try to set any photo as album cover photo in starred folder

### Diagnostics [REQUIRED]

    Diagnostics
    -----------
        Warning: zend.assertions is disabled although Lychee is in debug mode. For easier debugging code generation for assertions should be enabled.
         Foreign key: access_permissions.user_id     → users.id             : CASCADE
         Foreign key: albums.id                      → base_albums.id       : NO ACTION
         Foreign key: base_albums.owner_id           → users.id             : NO ACTION
         Foreign key: jobs_history.owner_id          → users.id             : NO ACTION
         Foreign key: oauth_credentials.user_id      → users.id             : CASCADE
         Foreign key: photos.album_id                → albums.id            : NO ACTION
         Foreign key: photos.owner_id                → users.id             : NO ACTION
         Foreign key: size_variants.photo_id         → photos.id            : NO ACTION
         Foreign key: sym_links.size_variant_id      → size_variants.id     : NO ACTION
         Foreign key: tag_albums.id                  → base_albums.id       : NO ACTION
        Info: Found 24 small that could be generated.
              You can use `php artisan lychee:generate_thumbs small 24` to generate them.
        Info: Found 823 medium that could be generated.
              You can use `php artisan lychee:generate_thumbs medium 823` to generate them.
        Info: Found 47571 small images without filesizes.
              You can use `php artisan lychee:variant_filesize 47571` to compute them.
    
    System Information
    ------------------
        Lychee Version (git):                    master (2f78c38) -- Up to date (5 days ago).
        DB Version:                              5.3.0
        
        composer install:                        --no-dev
        APP_ENV:                                 production
        APP_DEBUG:                               true
        APP_URL:                                 set
        APP_DIR:                                 default
        LOG_VIEWER_ENABLED:                      true
        LIVEWIRE_ENABLED:                        true
        
        System:                                  Linux
        PHP Version:                             8.3.6
        PHP User agent:                          Lychee/4 (https://lycheeorg.github.io/)
        Timezone:                                America/Boise
        Max uploaded file size:                  1000M
        Max post size:                           1000M
        Livewire chunk size:                     12.00 MB
        Max execution time:                      2000
        SQLite Version:                          3.37.2
        
        exec() Available:                        yes
        Imagick Available:                       1
        Imagick Enabled:                         1
        Imagick Version:                         1691
        GD Version:                              2.3.3
        Number of foreign key:                   10 found.
    
    Config Information
    ------------------
        version:                                 050300
        check_for_updates:                       1
        sorting_photos_col:                      title
        sorting_photos_order:                    ASC
        sorting_albums_col:                      title
        sorting_albums_order:                    DESC
        imagick:                                 1
        skip_duplicates:                         1
        small_max_width:                         0
        small_max_height:                        360
        medium_max_width:                        1920
        medium_max_height:                       1080
        lang:                                    en
        image_overlay_type:                      desc
        default_license:                         none
        compression_quality:                     90
        grants_full_photo_access:                1
        delete_imported:                         0
        mod_frame_enabled:                       1
        mod_frame_refresh:                       30
        thumb_2x:                                1
        small_2x:                                1
        medium_2x:                               1
        landing_page_enable:                     0
        force_migration_in_production:           0
        search_public:                           0
        hide_version_number:                     0
        SL_enable:                               0
        SL_for_admin:                            0
        recent_age:                              1
        grants_download:                         1
        photos_wraparound:                       1
        map_display:                             1
        zip64:                                   1
        map_display_public:                      0
        map_provider:                            Wikimedia
        force_32bit_ids:                         0
        map_include_subalbums:                   1
        update_check_every_days:                 100
        has_exiftool:                            0
        share_button_visible:                    1
        import_via_symlink:                      0
        has_ffmpeg:                              1
        location_decoding:                       1
        location_decoding_timeout:               30
        location_show:                           1
        location_show_public:                    0
        rss_enable:                              0
        rss_recent_days:                         7
        rss_max_items:                           100
        prefer_available_xmp_metadata:           0
        editor_enabled:                          1
        lossless_optimization:                   0
        swipe_tolerance_x:                       150
        swipe_tolerance_y:                       250
        local_takestamp_video_formats:           .avi|.mov
        log_max_num_line:                        1000
        unlock_password_photos_with_url_param:   0
        nsfw_visible:                            1
        nsfw_blur:                               0
        nsfw_warning:                            0
        nsfw_warning_admin:                      0
        nsfw_banner_override:                    
        map_display_direction:                   1
        album_subtitle_type:                     oldstyle
        upload_processing_limit:                 4
        new_photos_notification:                 0
        legacy_id_redirection:                   1
        zip_deflate_level:                       6
        SA_enabled:                              1
        default_album_protection:                1
        album_decoration:                        layers
        album_decoration_orientation:            row
        allow_username_change:                   1
        auto_fix_orientation:                    1
        use_job_queues:                          0
        random_album_id:                         starred
        use_last_modified_date_when_no_exif_date: 0
        layout:                                  justified
        date_format_photo_thumb:                 M j, Y, g:i:s A e
        date_format_photo_overlay:               M j, Y, g:i:s A e
        date_format_sidebar_uploaded:            M j, Y, g:i:s A e
        date_format_sidebar_taken_at:            M j, Y, g:i:s A e
        date_format_hero_min_max:                F Y
        date_format_hero_created_at:             M j, Y, g:i:s A T
        date_format_album_thumb:                 M Y
        upload_chunk_size:                       0
        nsfw_banner_blur_backdrop:               0
        search_pagination_limit:                 1000
        search_minimum_length_required:          4
        photo_layout_justified_row_height:       320
        photo_layout_masonry_column_width:       300
        photo_layout_grid_column_width:          250
        photo_layout_square_column_width:        200
        photo_layout_gap:                        12
        display_thumb_album_overlay:             always
        display_thumb_photo_overlay:             hover
        default_album_thumb_aspect_ratio:        1/1
        use_album_compact_header:                0
        login_button_position:                   left
        back_button_enabled:                     0
        back_button_text:                        Return to Home
        current_job_processing_visible:          1
    

### Browser & System [REQUIRED]

Windows 10, Chrome

### Please confirm (incomplete submissions will not be addressed)

- [X] I have provided easy and step-by-step instructions to reproduce the bug.
- [X] I understand my bug report will be removed if I haven't met the criteria above.

**Comment by ildyria** [comment]
> Trying to set any photo as album cover photo in starred folder gives the error: "No query results for model [App\Models\Album] starred"

Well... that is not supposed to work for smart albums :')
So I guess I messed up by leaving that option available. (sorry)

**Comment by hess8** [comment]
No problem.  Maybe it could be a random photo each time instead of the same
one every time.

**Reply by ildyria** [reply]
https://github.com/LycheeOrg/Lychee/pull/2429

**Reply by ildyria** [reply]
and #2430

**Comment by hess8** [comment]
Nice!
